yarpl/include/yarpl/observable/ObservableOperator.h

```diff
@@ -199,8 +199,13 @@ class MapOperator : public ObservableOperator<U, D, MapOperator<U, D, F>> {
         : SuperSub(std::move(observable), std::move(observer)) {}
 
     void onNext(U value) override {
-      auto& map = SuperSub::getObservableOperator();
-      SuperSub::observerOnNext(map->function_(std::move(value)));
+      try {
```

Comment:
Should we check `isTerminated` in here? As, even though previous onNext might have thrown an exception, it doesn't force onNext to be not called again, isn't it?

Reply:
Hmm yeah.  That's an issue with all the `FooOperator` classes in general.  They don't defend against getting follow-up `onNext()` callbacks after sending a `cancel()` to the upstream.

I think it's fine to wait for `SafeSubscriber/SubscriberBase` to deal with that, instead of duplicating the effort here.